src/unity/lib/unity_sframe.cpp

```diff
@@ -844,20 +844,50 @@ std::shared_ptr<unity_sframe_base> unity_sframe::append(
     if (this->num_columns() != other_sframe->num_columns()) {
       log_and_throw("Two SFrames have different number of columns");
     }
-    auto column_names = this->column_names();
-    auto other_column_names = other_sframe->column_names();
+    std::vector<std::string> column_names = this->column_names();
+    std::vector<std::string> other_column_names = other_sframe->column_names();
+    
+    size_t num_columns = column_names.size();
+
+    if(column_names != other_column_names) {
+      std::sort(column_names.begin(), column_names.end());
+      std::sort(other_column_names.begin(), other_column_names.end());
+
+      if(column_names != other_column_names) { 
+
+      std::vector<std::string> in_this, in_other;
+
+      std::set_difference(column_names.begin(), column_names.end(),
+                          other_column_names.begin(), other_column_names.end(),
+                          std::inserter(in_this, in_this.begin()));
+
+      std::set_difference(other_column_names.begin(), other_column_names.end(),
+                          column_names.begin(), column_names.end(),
+                          std::inserter(in_other, in_other.begin()));
+
+        std::ostringstream ss; 
+          ss << "Error: Attempting to append SFrame with columns "
+            << in_other << " to SFrame with columns " << in_this << ".";
```

Comment:
Do we want to use the set difference for `in_this`? I would expect as a user to get the full list of columns of the SFrame being appended to (including the intersection with `in_other`).

Reply:
Or maybe simplify it to the same message style as for types below (just show the names of columns in the `other_sframe` that aren't valid in `this` one)?

Reply:
Fair point; this just reports the mismatch.   What about this?  "Columns [....] in base SFrame not found in appending SFrame."  ?

Reply:
That sounds good.